### PR TITLE
Collinearity Speed Up v2

### PR DESCRIPTION
**Pull request by ChadGueli**
# Summary
Again, this is a major speed up. It eliminates recomputation along rows, saving a huge amount of time.

The VIFs are computed using the correlation formula, but optimized with only the diagonal inverted.

# Tests
It passes all of the unittest tests. But the `test_vif_calculation` test is removed. I would be happy to add another that checks the current formulation.

cc:
@tommydangerous @skunichetty

### CODE REVIEW

**Comment by skunichetty** [comment]
Sorry for the late response, is there any reason you switched back to VIFs from VDFs?

**Comment by ChadGueli** [comment]
Yeah, I took a closer look at the tests, and realized that in cases of perfect collinearity, the VDPs are uniformly split between the columns. This creates a problem if you have more than 3 perfectly collinear columns. Initially I thought that nothing is ever perfectly collinear so it doesn't matter, but then I realized that the test situation is highly plausible especially if someone doesn't know what they're doing or doesn't understand there data.

If you use VDPs, I think its best to use them in conjunction with another indicator of collinearity. The VDP formula and VIF formula do have a lot in common, so it could be wise to use them in conjunction.

File: mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py
Comment (skunichetty) [inline mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py:47]: Nitpick: Can you add the newline back in at the end of file?

**Review by skunichetty — COMMENTED** [review]
Code looks great! Once you resolve my comments I can approve for merging, or if you want to investigate a joint approach using VDPs you can add on as well.

File: mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py
Comment (ChadGueli) [reply mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py:25]: Yeah, but interesting that it worked.

File: mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py
Comment (ChadGueli) [reply mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py:26]: Good catch!

File: mage_ai/tests/data_cleaner/cleaning_rules/test_remove_collinear_columns.py
Comment (skunichetty) [reply mage_ai/tests/data_cleaner/cleaning_rules/test_remove_collinear_columns.py:434]: Agreed, albeit slightly annoying to code, it would be generally more descriptive to check for the subset of the features rather than their order.

**Comment by ChadGueli** [comment]
Yes, it should be.

**Comment by skunichetty** [comment]
It's been a while, but it should be good - i'll give it the go ahead.

**Review by skunichetty — APPROVED** [review]

**Pull request merged**
